### PR TITLE
fix(argo-workflows): Set only used values on SSO configuration

### PR DESCRIPTION
**Pull request by yu-croco**
I fixed this https://github.com/argoproj/argo-helm/pull/1473/files#r976930509 . 🙇 

---

Signed-off-by: yu-croco <yu.croco@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/values.yaml [hunk 1]
```diff
@@ -482,6 +482,11 @@ server:
     # redirectUrl: https://argo/oauth2/callback
     # rbac:
     #   enabled: true
+    ## When present, restricts secrets the server can read to a given list.
+    ## You can use it to restrict the server to only be able to access the
+    ## service account token secrets that are associated with service accounts
+    ## used for authorization.
+    #   secretWhitelist: []
```
Comment (yu-croco) [inline charts/argo-workflows/values.yaml:489]: I'm sorry, it is still used. 🙇 
ref: https://github.com/argoproj/argo-helm/pull/1473/files#r976930509

File: charts/argo-workflows/templates/controller/workflow-controller-config-map.yaml [hunk 2]
```diff
@@ -119,7 +119,38 @@ data:
...
+      {{- end }}
+      {{- if .customGroupClaimName }}
+      customGroupClaimName: {{ .customGroupClaimName }}
+      {{- end }}
+      {{- if .UserInfoPath }}
+      UserInfoPath: {{ .UserInfoPath }}
+      {{- end }}
+      {{- if .insecureSkipVerify }}
+      insecureSkipVerify: {{ .insecureSkipVerify }}
+      {{- end }}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/controller/workflow-controller-config-map.yaml:153]: I set only valid values.
ref: ref: https://github.com/argoproj/argo-workflows/blob/v3.4.0/config/sso.go#L10-L24

**Comment by vladlosev** [comment]
I wonder, what is the goal of this change? It does not enable any new features but on the downside, the there is an increased maintenance burden - the chart now has to keep up with  any changes in the SSOConfig struct.

**Comment by yu-croco** [comment]
Hi @vladlosev , thank you for checking PR!
This is because argo-workflow-configmap does not accept unsupported values from v3.4.0, like below
ref: https://argoproj.github.io/argo-workflows/upgrading/#feat-fail-on-invalid-config-8295

> The workflow controller will error on start-up if incorrectly configured, rather than silently ignoring mis-configuration.
> Failed to register watch for controller config map: error unmarshaling JSON: while decoding JSON: json: unknown field \"args\"


We set values like the below on `workflow-controller-config-map.yaml`, but there are unsupported values like `secretWhitelist` for configmap.
For keeping backward compatibility, I set only valid values on `workflow-controller-config-map.yaml` .

```yaml
{{- with .Values.server.sso }}
sso: {{- toYaml . | nindent 6 }}
...
```

I'm wondering if there are better ways... 🤔

**Review by vladlosev — APPROVED** [review]
That's a good reason. Thanks for explaining!

**Pull request merged**